src/components/elements/Button.astro

```diff
@@ -5,9 +5,11 @@ interface Props {
   style: "primary" | "secondary";
   size: "md" | "lg";
   fullWidth?: boolean;
+  icon?: string;
+  iconPosition?: "left" | "right";
 }
 
-const { href, label, style, size, fullWidth } = Astro.props;
+const { href, label, style, size, fullWidth, icon, iconPosition = "left" } = Astro.props;
```

Comment:
do we need to declare a value for the icon position here?

Reply:
This is a default value, in a case where an icon is present and no position is specified, it takes the default value here (left)